File: Framework/LanguageSupport/cpp/MMICPP/Adapter/SessionTools.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Niclas Delfs, Stephan Adam

#include "SessionTools.h"
#include <boost/algorithm/string.hpp>
 
using namespace MMIStandard;
 std::vector<string> SessionTools::GetSplittedIds(const string &sessionID)
{
	if (sessionID.empty()|| sessionID.compare("")==0)
	{
		 throw std::runtime_error("SessionId is empty");
	}
	std::vector<std::string>splitted;
	boost::split(splitted, sessionID, boost::is_any_of(":"));

	if (splitted.size() == 2 && !splitted[0].empty() && !splitted[1].empty())
	{
		return splitted;
	}
	else
	{
		return std::vector<std::string>{sessionID, "0"};
	}
	
}

File: Framework/LanguageSupport/java/MMIJava/src/Adapter/SessionTools.java
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Felix Gaisbauer

package Adapter;

public class SessionTools {
    /*
		Helper class for sessions
	*/

    // returns the scene id and the avatar id based on the session id
    public static String[] getSplittedIDs(String sessionId) throws RuntimeException {

        if (sessionId == null || sessionId.equals("")) {
            throw new RuntimeException("SessionId is empty");
        }

        String[] splitted = sessionId.split(":");

        // Test if the Format was correct
        if (splitted.length == 2 && !splitted[0].equals("") && !splitted[1].equals("")) {
            return splitted;
        } else
            return new String[]{sessionId, "0"};
    }
}
